haskell.nix/flake.nix

```diff
@@ -0,0 +1,114 @@
+{
+  nixConfig = {
+    flake-registry = "https://github.com/serokell/flake-registry/raw/master/flake-registry.json";
+  };
+
+  inputs = {
+    flake-compat = {
+      flake = false;
+    };
+    haskell-nix = {
+      inputs.hackage.follows = "hackage";
+      inputs.stackage.follows = "stackage";
+    };
+    hackage = {
+      flake = false;
+    };
+    stackage = {
+      flake = false;
+    };
+
+    # if you want to run weeder:
+    ##haskell-nix-weeder = {
+    ##  flake = false;
+    ##};
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, hackage, stackage, flake-compat }:
+
+  # for weeder:
+  ##outputs = { self, nixpkgs, haskell-nix, hackage, stackage, flake-compat, haskell-nix-weeder }:
+
+    let
+      pkgs = nixpkgs.legacyPackages.x86_64-linux.extend haskell-nix.overlay;
+      lib = pkgs.lib;
+
+      # invoke haskell.nix
+      hs-pkgs = pkgs.haskell-nix.stackProject {
+        src = pkgs.haskell-nix.haskellLib.cleanGit {
+          name = "pataq-package";
+          src = ./.;
+        };
+
+        # haskell.nix configuration
+        modules = [{
+          packages.pataq-package = {
```

Comment:
I'm still not so sure about using `pataq` everywhere instead of specifying the `packageName` once and then using it as `${packageName}` or something, but if you are sure it's better than whatever

Reply:
It does seem nice to have to replace the package name only once, but there is another difficulty that we also need people to update the name for the test component, I think there's no really convenient way to derive it. So I've changed flake.nix to have package name specified only once, and added a notice about updating the name for the test component as well